pkg/utils/env/openv.go

```diff
@@ -0,0 +1,60 @@
+package env
+
+import (
+	"os"
+	"strings"
+
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"k8s.io/client-go/rest"
+
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	v1 "k8s.io/apimachinery/pkg/runtime/schema"
+	serializer "k8s.io/apimachinery/pkg/runtime/serializer"
+	runtime "k8s.io/apimachinery/pkg/runtime"
+)
+
+var log = logf.Log.WithName("env")
+
+func DetectOpenshift() (bool, error) {
```

Comment:
Please change the existing method... clients are either tied to a specific commit and unaffected, or this will force them to change their code, which is fine.

Reply:
added logic in existing method